Dockerfile

```diff
@@ -0,0 +1,23 @@
+############################
+# STEP 1 build executable binary
+############################
+FROM golang:alpine AS builder
+# Install git.
+# Git is required for fetching the dependencies.
+RUN apk update && apk add --no-cache git
+COPY . $GOPATH/src/insomniacslk/irc-slack
+WORKDIR $GOPATH/src/insomniacslk/irc-slack
+# Fetch dependencies.
+# Using go get.
+RUN go get -d -v
+# Build the binary.
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -installsuffix cgo -ldflags="-w -s" -o /go/bin/irc-slack
+
+############################
+# STEP 2 build a small image
+############################
+FROM scratch
+# Copy our static executable.
+COPY --from=builder /go/bin/irc-slack /go/bin/irc-slack
```

Comment:
Same source and destination, is this really required? Unless this is something that in Docker's language means something else?

Reply:
its copying from the first container (build container)'s /go/bin/irc-slack to the empty container /go/bin/irc-slack

So yea I think its necessary, i don't end up using the multi container stuff very often myself so not 100%